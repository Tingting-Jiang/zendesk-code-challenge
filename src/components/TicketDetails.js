import React from "react";


// const TicketDetails = ({ticket = {
//     requester_id: 100,
//     subject: "in nostrud occaecat consectetur aliquip",
//     description: "Esse esse quis ut esse nisi tempor sunt. Proident",
//     priority: "low",
//     status: "open"
// }, setList}) => {


const TicketDetails = ({ticket, setList}) => {
    
    const {subject,
        description,
        priority,
        status,
        created_at,
        submitter_id,
        requester_id,
        assignee_id} = ticket;
   
    return (
        ticket && (
            <>
                <button
                    className="btn btn-primary my-5"
                    onClick={() => setList(true)}>
                    Back to Tickets  List
                </button>
                <h3>Subject: {subject === null ? "Unknown" : subject}</h3>
    
                <table className="table">
                    <thead>
                    <tr>
                        <th scope="col-2">#</th>
                        <th scope="col-3">Item</th>
                        <th scope="col-7">Details</th>
                       
                    </tr>
                    </thead>
                    
                    <tbody>


                    <tr>
                        <th scope="row">1</th>
                        <td>Description</td>
                        <td>{description === null ? "Unknown" : description}</td>
                    </tr>

                    <tr>
                        <th scope="row">2</th>
                        <td>Priority</td>
                        <td>{priority === null ? "Unknown" : priority}</td>
                    </tr>

                    <tr>
                        <th scope="row">3</th>
                        <td>Status</td>
                        <td>{status === null ? "Unknown" : status}</td>
                    </tr>
                    
                    <tr>
                        <th scope="row">4</th>
                        <td>Created At</td>
                        <td>{created_at === null ? "Unknown" : created_at}</td>
                    </tr>
                    <tr>
                        <th scope="row">5</th>
                        <td>Submitter ID</td>
                        <td>{submitter_id === null ? "Unknown" : submitter_id}</td>
                    </tr>
                    <tr>
                        <th scope="row">6</th>
                        <td>Requester ID</td>
                        <td>{requester_id === null ? "Unknown" : requester_id}</td>
                    </tr>

                    <tr>
                        <th scope="row">7</th>
                        <td>Assignee ID</td>
                        <td>{assignee_id === null ? "Unknown" : assignee_id}</td>
                    </tr>
                    
                    
                    </tbody>
            
                </table>
                
            
                {/*// <p>Status: {ticket.status}</p>*/}
                {/*// <p >Description: {ticket.description}</p>*/}
                {/*// <p>Priority: {ticket.priority} </p>*/}
            </>
        )
        
    );


};
export default TicketDetails;