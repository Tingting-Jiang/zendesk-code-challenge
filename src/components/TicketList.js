import React from 'react'



const TicketList = ({tickets, setList, setTicket}) => {
    
    // if (loading) {
    //     return <h2>Loading...</h2>;
    // }
    
    // console.log(tickets);
    
    const ViewTicket = (ticket) => {
        // console.log("event is", ticket);
        setTicket(ticket);
        setList(false);
        // console.log("after setting: ", ticket);
        
    }

    return (
        (tickets !== undefined) && (
        
        <ol className="list-group list-group-numbered mb-4">
            {tickets.map(ticketItem => (
              
                <li key={ticketItem.id}
                    
                    className=" card list-group-item list-group-item-action list-group-flush"
                    onClick={() => ViewTicket(ticketItem)}
                    id="list-item">

                        {ticketItem.subject}
                        <span className="float-end">Priority âž¡ {ticketItem.status === null ?
                    "Unknown" : ticketItem.status}</span>

                </li>
            ))}
        </ol>
        )
    
    )
}
export default TicketList;