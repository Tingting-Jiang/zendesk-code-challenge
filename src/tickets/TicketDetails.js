import React from "react";


const TicketDetails = ({ticket = {
    
    requester_id: 100,
    subject: "in nostrud occaecat consectetur aliquip",
    description: "Esse esse quis ut esse nisi tempor sunt. Proident",
    priority: "low",
    status: "open"
}, setList}) => {
    return (
        <>
            <button
                className="btn btn-primary my-5"
                onClick={() => setList(true)}>
                Back to Main Page
            </button>
            <h3>Subject: {ticket.subject}</h3>
            <p>Status: {ticket.status}</p>
            <p >Description: {ticket.description}</p>
            <p>Priority: {ticket.priority} </p>
        </>
    )


};
export default TicketDetails;