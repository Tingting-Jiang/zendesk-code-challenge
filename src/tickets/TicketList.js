import React, { useEffect } from 'react'
import { fetchTickets } from './fetchTickets'

const TicketList = ({tickets, setList, setTicket}) => {
    
    // if (loading) {
    //     return <h2>Loading...</h2>;
    // }
    
    // console.log(tickets);
    
    
    const ViewTicket = (ticket) => {
        // console.log("event is", ticket);
        setTicket(ticket);
        setList(false);
        console.log("after setting: ", ticket);
        
    }

    return (
        
        <ul className="list-group mb-4">
            {tickets.map(ticketItem => (
                <li key={ticketItem.id}
                    className="list-group-item"
                onClick={() => ViewTicket(ticketItem)}>
                    {ticketItem.subject}
                    <span className="d-none">
                        {ticketItem.requester_id}
                    </span>
                
                </li>
            ))}
        </ul>
    
    )
}
export default TicketList;