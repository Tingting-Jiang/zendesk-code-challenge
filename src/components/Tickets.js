import React, { useState } from 'react'
import TicketDetails from './TicketDetails'
import TicketList from './TicketList'

const Tickets = ({tickets, isList, setList, loading}) => {
    let [ticket, setTicket]= useState();

    if (loading) {
        return <h2>Loading...</h2>;
    }
    
    return (
   
        <div>
            {isList &&
                <TicketList tickets={tickets} setList={setList}
                            setTicket={setTicket}/> }
            
            {!isList &&
                <TicketDetails setList={setList}
                    ticket={ticket}/>}
        </div>
        
    );
    
    
}
export default Tickets;