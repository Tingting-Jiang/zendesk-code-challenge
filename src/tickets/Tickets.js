import React, { useEffect, useState } from 'react'
import { fetchTickets } from './fetchTickets'
import TicketDetails from './TicketDetails'
import TicketList from './TicketList'
import Pagination from './Pagination'

const Tickets = ({tickets, isList, setList}) => {
   

    // if (loading) {
    //     return <h2>Loading...</h2>;
    // }
    
    // console.log(tickets);
    
    
    // let [isList, setList] = useState(true);
    let [ticket, setTicket]= useState();
    
    return (
        <div>
            {isList &&
            <TicketList tickets={tickets} setList={setList} setTicket={setTicket}/> }
            
            {!isList &&
                <TicketDetails setList={setList}
                    ticket={ticket}/>}
        </div>
    )
    
    
    
    
    // return (
    //
    //     <ul className="list-group mb-4">
    //         {tickets.map(ticket => (
    //             <li key={ticket.id} className="list-group-item">
    //
    //
    //
    //                 {ticket.subject}
    //
    //             </li>
    //         ))}
    //     </ul>
    //
    // )
}
export default Tickets;