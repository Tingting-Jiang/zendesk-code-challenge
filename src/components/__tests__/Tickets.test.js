import React from 'react';
import Tickets from '../Tickets'
import {shallow } from"enzyme";
import testData from "../../data/test.json";
import TicketList from '../TicketList';
import TicketDetails from '../TicketDetails'


describe("<Tickets/>", () => {
    
    it("render Tickets without data", () => {
        const wrapper = shallow(<Tickets />);
        expect(wrapper).toMatchSnapshot();
    });
    
    
    it ("render Tickets with loading to be true", () =>{
        const wrapper = shallow(<Tickets loading={true}/>);
        expect(wrapper.find('h2').text()).toBe("Loading...");
   
    })
    
    
    it ("render Tickets with TicketList", () =>{
        const ticket = testData;
        const setList = jest.fn();
        const wrapper = shallow(<Tickets tickets={ticket}
                                         isList={true}
                                         setList={setList}
                                         loading={false}/>);
        expect(wrapper.find(TicketList)).toHaveLength(1);
        expect(wrapper.find(TicketDetails)).toHaveLength(0);

    })
    
    
    it ("render Tickets with TicketDetails", () =>{
        const ticket = testData;
        const setList = jest.fn();
        const wrapper = shallow(<Tickets tickets={ticket}
                                         isList={false}
                                         setList={setList}
                                         loading={false}/>);
        expect(wrapper.find(TicketList)).toHaveLength(0);
        expect(wrapper.find(TicketDetails)).toHaveLength(1);
        
    })
})