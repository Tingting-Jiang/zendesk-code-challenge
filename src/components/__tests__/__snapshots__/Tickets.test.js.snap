// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tickets/> render Tickets without data 1`] = `ShallowWrapper {}`;
